src/components/TableComponent/index.jsx

```diff
@@ -1,166 +1,120 @@
-      }
-    }
-  }
-
-  componentWillUnmount() {
-    this.props.clearTable();
-  }
-
-  renderTableContent() {
-    const {
-      className,
-      currentPage,
-      pageCount,
-      tableSortable,
-      data,
-      ordering,
-      formatData,
-      id,
-      loading,
-      enterpriseId,
-    } = this.props;
-
-    const sortByColumn = (column, direction) => {
+const TableComponent = ({
+  className, currentPage, pageCount, tableSortable, data, formatData, id, loading, enterpriseId, totalCount,
+  paginateTable, clearTable, columns, error, sortTable,
+}) => {
+  const sortByColumn = useCallback(args => {
+    const { sortBy } = args;
+    if (sortBy && sortBy.length > 0) {
+      const column = sortBy[0];
+      const ordering = `${column.desc ? '-' : ''}${column.id}`;
       updateUrl({
         page: 1,
-        ordering: direction === 'desc' ? `-${column.key}` : column.key,
+        ordering,
       });
+      sortTable(ordering);
       sendEnterpriseTrackEvent(enterpriseId, 'edx.ui.enterprise.admin_portal.table.sorted', {
         tableId: id,
-        column: column.label,
-        direction,
+        column: column.id,
+        direction: column.desc ? 'desc' : 'asc',
       });
-    };
-
-    const columnConfig = this.props.columns.map(column => ({
-      ...column,
-      onSort: column.columnSortable ? (direction) => sortByColumn(column, direction) : null,
-    }));
-
-    let sortDirection;
-    let sortColumn;
-
-    if (tableSortable) {
-      sortDirection = ordering && ordering.indexOf('-') !== -1 ? 'desc' : 'asc';
-      sortColumn = (ordering && ordering.replace('-', '')) || columnConfig[0].key;
     }
+  }, []);
 
-    return (
-      <div className={className}>
-        <div className="row">
-          <div className="col">
-            {loading && <TableLoadingOverlay />}
-            <div className="table-responsive">
-              <Table
-                id={id}
-                className="table-sm table-striped"
-                columns={columnConfig}
-                data={formatData(data)}
-                tableSortable={tableSortable}
-                defaultSortedColumn={sortColumn}
-                defaultSortDirection={sortDirection}
-              />
-            </div>
-          </div>
-        </div>
-        <div className="row mt-2">
-          <div className="col d-flex justify-content-center">
-            <Pagination
-              paginationLabel={`${id}-pagination`}
-              pageCount={pageCount}
-              currentPage={currentPage}
-              onPageSelect={(page) => {
-                updateUrl({ page });
-                sendEnterpriseTrackEvent(enterpriseId, 'edx.ui.enterprise.admin_portal.table.paginated', {
-                  tableId: id,
-                  page,
-                });
-              }}
-            />
+  useEffect(() => {
+    // Get initial data
+    paginateTable();
+    return clearTable;
+  }, []);
+
+  const handlePageChange = (page) => {
```

Comment:
As mentioned else, pagination changes for `DataTable` should also come through the `fetchData` callback function rather than our existing custom implementation.